File: 백준/Silver/1244. 스위치 켜고 끄기/스위치 켜고 끄기.js
const fs = require("fs");
const stdin = (
  process.platform === "linux"
    ? fs.readFileSync("/dev/stdin").toString()
    : fs.readFileSync("./input.txt").toString()
).split("\n");

const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

const N = Number(input());

const state = input()
  .trim()
  .split(" ")
  .map((v) => Number(v));

const studentNumber = Number(input());
const students = [];

for (let i = 0; i < studentNumber; i++) {
  students.push(
    input()
      .trim()
      .split(" ")
      .map((v) => Number(v))
  );
}

function man(n) {
  for (let i = 1; i * n - 1 < N; i++) {
    state[n * i - 1] = state[n * i - 1] ? 0 : 1;
  }
}

function woman(n) {
  const idx = n - 1;
  let i = 1;
  state[idx] = state[idx] ? 0 : 1;
	while (i < N) {
		if (idx - i >= 0 && idx + i < N && state[idx - i] === state[idx + i]) {
      state[idx - i] = state[idx - i] ? 0 : 1;
      state[idx + i] = state[idx + i] ? 0 : 1;
    } else {
      return;
    }
    i++;
  }
}

students.forEach((v) => {
  if (v[0] === 1) man(v[1]);
  else woman(v[1]);
});

let arrIdx = -1;
const answer = Array.from(Array(Math.floor(N / 20) + 1), () => Array(0));

state.forEach((v, idx) => {
  if (idx % 20 === 0) arrIdx++;
  answer[arrIdx].push(v);
});

for (const a of answer) {
  console.log(a.join(" "));
}

File: 백준/Silver/1244. 스위치 켜고 끄기/스위치 켜고 끄기.java
import java.io.*;
import java.util.*;

public class Main {

	static int[] switches;
	static int N;

	static void man(int idx) {

		for (int i = idx; i <= N; i++) {
			if ((i) % idx == 0) {
				switches[i - 1] = Math.abs(switches[i - 1] - 1);
			}
		}
	}

	static void woman(int idx) {
		int f = idx, l = idx;

		while (f >= 0 && l < N) {
			if (switches[f] != switches[l]) {
				break;
			}
			f--;
			l++;
		}
		
		f++;
		l--;
		
		for (int i = f; i <= l; i++) {
			switches[i] = Math.abs(switches[i] - 1);
		}
		
		return;
	}

	public static void main(String[] args) throws Exception {
		//System.setIn(new FileInputStream("res/input.txt"));
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		N = Integer.parseInt(br.readLine());

		StringTokenizer st = new StringTokenizer(br.readLine(), " ");

		switches = new int[N];

		for (int n = 0; n < N; n++) {
			switches[n] = Integer.parseInt(st.nextToken());
		}

		int S = Integer.parseInt(br.readLine());

		for (int s = 0; s < S; s++) {
			StringTokenizer str = new StringTokenizer(br.readLine(), " ");
			int sex = Integer.parseInt(str.nextToken());
			int num = Integer.parseInt(str.nextToken());
			if (sex == 1) {
				man(num);
			} else {
				woman(num - 1);
			}
		}
		
		String str = "";
		
		for (int i = 1; i <= N; i ++) {
			str += Integer.toString(switches[i-1]);
			if (i % 20 == 0) {
				str += "\n";
			} else {
				str += " ";
			}
		}
		
		System.out.println(str.trim());

		br.close();
	}
}